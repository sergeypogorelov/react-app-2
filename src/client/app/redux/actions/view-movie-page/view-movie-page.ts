import { GLOBALS } from '../../../core/constants/globals';
import {
  LOAD_MOVIE,
  LOAD_MOVIE_PENDING,
  LOAD_MOVIE_FULFILLED,
  LOAD_MOVIE_REJECTED,
  LOAD_MOVIES_BY_GENRE,
  LOAD_MOVIES_BY_GENRE_FULFILLED,
  LOAD_MOVIES_BY_GENRE_PENDING,
  LOAD_MOVIES_BY_GENRE_REJECTED,
} from '../action-types';

import { SearchBy } from '../../../core/enums/search-by.enum';

import { Movie } from '../../../core/interfaces/movie/movie.interface';
import { MoviesResponse } from '../../../core/interfaces/movie/movies-response.interface';

import { MoviesApi } from '../../../core/api/movies/movies-api.class';

export const loadMovie = (id: number): LoadMovieAction => {
  return {
    type: LOAD_MOVIE,
    payload: MoviesApi.getById(id),
  };
};

export const loadMovieByGenre = (genre: string): LoadMoviesByGenreAction => {
  return {
    type: LOAD_MOVIES_BY_GENRE,
    payload: MoviesApi.getAll({
      search: genre,
      searchBy: SearchBy.Genres,
      limit: GLOBALS.moviesPerPage,
    }),
  };
};

export interface LoadMovieAction {
  type: typeof LOAD_MOVIE;
  payload: Promise<Movie>;
}

export interface LoadMoviePendingAction {
  type: typeof LOAD_MOVIE_PENDING;
}

export interface LoadMovieFulfilledAction {
  type: typeof LOAD_MOVIE_FULFILLED;
  payload: Movie;
}

export interface LoadMovieRejectedAction {
  type: typeof LOAD_MOVIE_REJECTED;
}

export interface LoadMoviesByGenreAction {
  type: typeof LOAD_MOVIES_BY_GENRE;
  payload: Promise<MoviesResponse>;
}

export interface LoadMoviesByGenrePendingAction {
  type: typeof LOAD_MOVIES_BY_GENRE_PENDING;
}

export interface LoadMoviesByGenreFulfilledAction {
  type: typeof LOAD_MOVIES_BY_GENRE_FULFILLED;
  payload: MoviesResponse;
}

export interface LoadMoviesByGenreRejectedAction {
  type: typeof LOAD_MOVIES_BY_GENRE_REJECTED;
}

export type ViewMoviePageActionTypes =
  | LoadMovieAction
  | LoadMoviePendingAction
  | LoadMovieFulfilledAction
  | LoadMovieRejectedAction
  | LoadMoviesByGenreAction
  | LoadMoviesByGenrePendingAction
  | LoadMoviesByGenreFulfilledAction
  | LoadMoviesByGenreRejectedAction;
