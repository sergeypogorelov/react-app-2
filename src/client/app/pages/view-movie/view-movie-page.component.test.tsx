import React from 'react';
import { Provider } from 'react-redux';
import renderer from 'react-test-renderer';
import configureStore from 'redux-mock-store';

import { ViewMoviePage } from './view-movie-page.component';

jest.mock('react-router-dom', () => ({
  ...jest.requireActual('react-router-dom'),
  useParams: () => ({
    movieId: 1,
  }),
}));

const mockStore = configureStore();

test('MovieDetails component should render correctly', () => {
  const store = mockStore({
    viewMoviePage: {
      movie: null,
      moviesByGenre: [],
    },
    searchMoviePage: {
      search: '',
    },
  });

  const json = renderer
    .create(
      <Provider store={store}>
        <ViewMoviePage />
      </Provider>
    )
    .toJSON();

  expect(json).toMatchSnapshot();
});
