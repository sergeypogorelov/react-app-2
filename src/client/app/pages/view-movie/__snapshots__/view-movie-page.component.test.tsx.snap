// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MovieDetails component should render correctly 1`] = `
<div
  className="search-movie-page"
>
  <div
    className="header"
  >
    <div
      className="header-logo"
    >
      <p
        className="title-main m-0"
      >
        <span
          className="title-main-prefix"
        >
          netflix
        </span>
        <span>
          roulette
        </span>
      </p>
    </div>
    
    <div
      className="header-content"
    />
  </div>
  <div
    className="info-container d-flex align-items-center justify-content-between"
  />
  <div
    className="movies-list"
  >
    <h3
      className="movies-list-empty"
    >
      No movies found
    </h3>
  </div>
</div>
`;
